pkg/helpers/helper.go

```diff
@@ -1,8 +1,45 @@
 package helpers
 
-import "strings"
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
 
 // IsPreReleaseVersion checks if the given version is a Terraform pre-release version
 func IsPreReleaseVersion(version string) bool {
 	return strings.Contains(version, "-alpha") || strings.Contains(version, "-beta") || strings.Contains(version, "-rc")
 }
+
+// FindRequiredVersionInFile finds the required Terraform version in a given .tf file (using required_version = ">= x.x.x")
+func FindRequiredVersionInFile(filepath string) string {
+	bytes, err := os.ReadFile(filepath)
+	if err != nil {
+		fmt.Printf("Unable to find version number in file: %s", filepath)
+		return ""
+	}
+	re := regexp.MustCompile(`>=\s(\d+\.\d+\.\d+)`)
```

Comment:
Can't this be a provider version too by accident? Would it be suitable to perhaps use hclparse for this?

Reply:
Hashicorp has a package called `go-version/constraint` that we might be able to use. For now I was just trying to make it work. I did update the regex to look for `required_version = "(MATCH_HERE)"`.